src/app/routes/article/getInitialData/addMpuBlock/index.js

```diff
@@ -0,0 +1,108 @@
+import path from 'ramda/src/path';
+import splitAt from 'ramda/src/splitAt';
+import flatten from 'ramda/src/flatten';
+import clone from 'ramda/src/clone';
+import { ARTICLE_PAGE } from '#app/routes/utils/pageTypes';
+import isLive from '#app/lib/utilities/isLive';
+
+const mpuBlock = {
+  type: 'mpu',
+  model: {},
+};
+
+const insertMpuBlock = (blocks, insertIndex) => {
+  if (!insertIndex) {
+    return blocks;
+  }
+
+  const newBlocks = clone(blocks);
+
+  const textBlocksWithParagraphs = [];
+
+  blocks.forEach((block, i) => {
+    if (block.type === 'text') {
```

Comment:
Just thinking, is it possible to have a article page/story page without any text blocks? If so, might be worth having a unit test for this.